src/Compilers/CSharp/Portable/Binder/Binder_Expressions.cs

```diff
@@ -2278,16 +2283,15 @@ private bool RefMustBeObeyed(bool isDelegateCreation, ArgumentSyntax argumentSyn
                 {
                     case BoundKind.PropertyAccess:
                     case BoundKind.IndexerAccess:
-                        return CheckIsVariable(argumentSyntax, arg, BindValueKind.RefOrOut, false, diagnostics);
+                        CheckIsVariable(argumentSyntax, arg, BindValueKind.RefOrOut, false, diagnostics);
+                        return;
```

Comment:
Ok, this doesn't really matter, as this logic is only applied in the case of a single argument, and the purpose of `hadError` is to avoid cascading diagnostics while scanning across multiple arguments.

---
In reply to: [122488321](https://github.com/dotnet/roslyn/pull/20121#discussion_r122488321) [](ancestors = 122488321)